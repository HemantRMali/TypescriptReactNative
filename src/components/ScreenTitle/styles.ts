import {StyleSheet} from 'react-native';
export default StyleSheet.create({
  screenTitle: {
    alignSelf: 'center',
    margin: 20,
    fontSize: 40,
    color: '#595856',
  },
});
