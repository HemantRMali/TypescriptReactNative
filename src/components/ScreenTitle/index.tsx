import React from 'react';
import {Text} from 'react-native';
import styles from './styles';

interface ScreenTitleProps {
  title: string;
  style?: object;
  tesId?: string;
}
const ScreenTitle: React.FC<ScreenTitleProps> = (props) => {
  return (
    <Text testID={props.tesId} style={[styles.screenTitle, props.style]}>
      {props.title}
    </Text>
  );
};

export default ScreenTitle;
