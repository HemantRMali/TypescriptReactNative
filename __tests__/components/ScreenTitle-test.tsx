// __tests__/ScreenTitle-test.js
import React from 'react';
import ScreenTitle from '../../src/components/ScreenTitle/ScreenTitle';
import {render} from '@testing-library/react-native';

describe('ScreenTitle', () => {
  it('renders properly', () => {
    const {toJSON} = render(<ScreenTitle title="Test Title" />);

    expect(toJSON()).toMatchSnapshot();
  });
});
