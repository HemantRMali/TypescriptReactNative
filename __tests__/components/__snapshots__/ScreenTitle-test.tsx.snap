// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenTitle renders properly 1`] = `
<Text
  style={
    Array [
      Object {
        "alignSelf": "center",
        "color": "#595856",
        "fontSize": 40,
        "margin": 20,
      },
      undefined,
    ]
  }
>
  Test Title
</Text>
`;
